components/Dashboard/ShiftsDetails.js

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect } from "react";
+import {
+  ShiftsContainer,
+  MorningShiftAccordion,
+  AfternoonShiftAccordion,
+  NightShiftAccordion,
+} from "@/components/Dashboard/Dashboard.styled";
+import NurseSlot from "./NurseSlot";
+import OneShift from "./OneShift";
+
+export default function ShiftDetails({
+  shifts,
+  onAddNurse,
+  onRemoveNurse,
+  nurseId,
+}) {
+  const [currentShiftType, setCurrentShiftType] = useState(null);
+  const [nursesList, setNursesList] = useState([]);
+
+  useEffect(() => {
+    async function fetchNurses() {
+      try {
+        const response = await fetch("/api/nurses");
+        const data = await response.json();
+        setNursesList(data);
+      } catch (error) {
+        console.error("Error fetching nurses:", error);
+      }
+    }
+
+    fetchNurses();
+  }, []);
```

Comment:
Is there a reason why you're not using the SWR hook here? When using fetch, you potentially make unnecessary requests. When using SWR, the hook makes sure not to overfetch even when it's called multiple times.